functions

```diff
@@ -77,16 +71,45 @@ service_exposed_ports() {
 service_link() {
   local APP="$2"
   local SERVICE="$1"
+  local SERVICE_URL=$(service_url "$SERVICE")
+  local SERVICE_NAME=$(get_service_name "$SERVICE")
   local SERVICE_ROOT="$PLUGIN_DATA_ROOT/$SERVICE"
+  local EXISTING_CONFIG=$(dokku config "$APP")
+  local DATABASE_NAME=$(get_database_name "$SERVICE")
+  local LINK=$(echo "$EXISTING_CONFIG" | grep "$PLUGIN_SCHEME://.*/$DATABASE_NAME" | cut -d: -f1) || true
+  local DEFAULT_ALIAS=$(echo "$EXISTING_CONFIG" | grep "${PLUGIN_DEFAULT_ALIAS}_URL") || true
+  local SERVICE_ALIAS=$(service_alias "$SERVICE")
   local LINKS_FILE="$SERVICE_ROOT/LINKS"
 
+  [[ -n $LINK ]] && dokku_log_fail "Already linked as $LINK"
   mkdir -p "$SERVICE_ROOT" || dokku_log_fail "Unable to create service directory"
   touch "$LINKS_FILE"
   echo "$APP" >> "$LINKS_FILE"
   sort "$LINKS_FILE" -u -o "$LINKS_FILE"
 
-  dokku_log_info1 "Restarting app $APP"
-  dokku ps:restart "$APP"
+  local ALIAS="$PLUGIN_DEFAULT_ALIAS"
+  if [[ -n $DEFAULT_ALIAS ]]; then
+    ALIAS=$(service_alternative_alias "$EXISTING_CONFIG")
+  fi
+  dokku docker-options:add "$APP" build,deploy,run "--link $SERVICE_NAME:$SERVICE_ALIAS"
+  dokku config:set "$APP" "${ALIAS}_URL=$SERVICE_URL"
+}
+
+service_alternative_alias() {
+  local EXISTING_CONFIG="$1"
+  local COLORS=(AQUA BLACK BLUE FUCHSIA GRAY GREEN LIME MAROON NAVY OLIVE PURPLE RED SILVER TEAL WHITE YELLOW)
```

Comment:
I think we’ll be in an infinite loop. I thought 16 database connections (per plugin) would be enough but we could add more :)